server/util/jwt.js

```diff
@@ -0,0 +1,69 @@
+const crypto = require('crypto');
+const jwt = require('jsonwebtoken');
+const expressjwt = require('express-jwt');
+const { UnauthorizedError } = require('express-jwt');
+const config = require('../../config/server');
+
+let secret = config.JWT_SECRET;
+if (!secret) {
+  secret = crypto
+    .randomBytes(64)
+    .toString('hex')
+    .slice(0, 64);
+}
+
+function getToken(req) {
+  if (
+    req.headers.authorization &&
+    req.headers.authorization.split(' ')[0] === 'Bearer'
+  ) {
+    return req.headers.authorization.split(' ')[1];
+  } else if (req.cookies && req.cookies.auth) {
+    return req.cookies.auth;
+  }
+  if (req.route.path === '/users/:wallet') {
+    return null;
+  }
+  throw new UnauthorizedError('credentials_required', {
+    message: 'No authorization token was found',
+  });
+}
+
+function setAuthHeader(res) {
+  res.setHeader('Surrogate-Control', 'no-store');
+  res.setHeader(
+    'Cache-Control',
+    'no-store, no-cache, must-revalidate, proxy-revalidate'
+  );
+  res.setHeader('Pragma', 'no-cache');
+  res.setHeader('Expires', '0');
+}
```

Comment:
ambiguous naming of `setAuthHeader` which is actually setting header to `no-cache`